src/eradiate/scenes/surface/_dem.py

```diff
@@ -1,50 +1,320 @@
 from __future__ import annotations
 
-import typing as t
-import warnings
-
 import attrs
+import mitsuba as mi
 import numpy as np
 import pint
 import xarray as xr
-from pinttr.util import ensure_units
 
 from ._core import Surface
-from ..bsdfs import BSDF, LambertianBSDF, bsdf_factory
-from ..core import InstanceSceneElement, NodeSceneElement
-from ..shapes import BufferMeshShape, FileMeshShape, Shape, shape_factory
+from ..bsdfs import LambertianBSDF, OpacityMaskBSDF
+from ..geometry import PlaneParallelGeometry, SceneGeometry, SphericalShellGeometry
+from ..shapes import (
+    BufferMeshShape,
+    FileMeshShape,
+    RectangleShape,
+    Shape,
+    SphereShape,
+    shape_factory,
+)
 from ...attrs import documented, get_doc, parse_docs
+from ...constants import EARTH_RADIUS
 from ...units import to_quantity
 from ...units import unit_context_config as ucc
+from ...units import unit_context_kernel as uck
 from ...units import unit_registry as ureg
 
 
-@parse_docs
-@attrs.define(eq=False, slots=False)
-class DEMSurface(Surface):
+def mesh_from_dem(
+    da: xr.DataArray,
+    geometry: SceneGeometry,
+    planet_radius: pint.Quantity | float = EARTH_RADIUS,
```

Comment:
Let's do something safer (and reject plain floats):
```suggestion
    planet_radius: pint.Quantity | None = None,
```